libraries/collectd_service.rb

```diff
@@ -113,11 +113,16 @@ def action_enable
             end
           end
 
+          pid_file =  if node['collectd']['service']['configuration']['pid_file']
```

Comment:
Isn't this being passed in as `new_resource.configuration['pid_file']`? We should use that instead.